major_event_log/views.py

```diff
@@ -18,32 +30,24 @@ def index(request):
 
 def event_details(request, event_id):
     """Loads the event details page of the event with the given id."""
-    try:
-        event = Event.objects.get(id=event_id)
-    except Event.DoesNotExist:
-        raise Http404
+    event = get_event_or_404(event_id)
     context = {'event': event}
     return render(request, 'major-event-log/event_details.html', context)
 
 
 def event_atom(request, event_id):
     """Loads the ATOM record for the event with the given id."""
-    try:
-        event = Event.objects.get(id=event_id)
-    except Event.DoesNotExist:
-        raise Http404
-    context = {'event': event,
-               'full_url': request.build_absolute_uri()[:-5] + '/'}
+    event = get_event_or_404(event_id)
+    event_detail_url = request.build_absolute_uri(
```

Comment:
Check this out. https://docs.djangoproject.com/en/1.8/ref/models/instances/#get-absolute-url

Then we will only need
```python
# ...
event_detail_url  = request.build_absolute_uri(event.get_absolute_url())
# ...
```